hamlpy/parser/nodes.py

```diff
@@ -48,6 +48,16 @@ def read_node(stream, prev, compiler):
         if indent:
             indent = indent[0] * len(indent)
 
+        # empty lines with carriage returns are recorded as newlines on previous node
+        # if followed by a newline, it is skipped
+        if stream.text[stream.ptr] == '\r':
```

Comment:
i think it would be easier to understand this if it was happening at the stream level.. what if we added `def is_newline()` and `def read_newline()` to the `Stream` class?